packages/svelte/src/compiler/phases/3-transform/client/visitors/shared/fragment.js

```diff
@@ -69,60 +97,60 @@ export function process_children(nodes, expression, is_element, { visit, state }
 				sequence = [];
 			}
 
-			if (
-				node.type === 'SvelteHead' ||
-				node.type === 'TitleElement' ||
-				node.type === 'SnippetBlock'
-			) {
-				// These nodes do not contribute to the sibling/child tree
-				// TODO what about e.g. ConstTag and all the other things that
-				// get hoisted inside clean_nodes?
-				visit(node, state);
+			let child_state = state;
+
+			if (is_static_element(node)) {
+				skipped += 1;
+			} else if (node.type === 'EachBlock' && nodes.length === 1 && is_element) {
+				node.metadata.is_controlled = true;
 			} else {
-				if (node.type === 'EachBlock' && nodes.length === 1 && is_element) {
-					node.metadata.is_controlled = true;
-					visit(node, state);
-				} else {
-					const id = get_node_id(
-						expression(false),
-						state,
-						node.type === 'RegularElement' ? node.name : 'node'
-					);
-
-					expression = (is_text) => b.call('$.sibling', id, is_text && b.true);
-
-					visit(node, {
-						...state,
-						node: id
-					});
-				}
+				const id = flush_node(false, node.type === 'RegularElement' ? node.name : 'node');
+				child_state = { ...state, node: id };
 			}
+
+			visit(node, child_state);
 		}
 	}
 
 	if (sequence.length > 0) {
-		// if the final item in a fragment is static text,
-		// we need to force `hydrate_node` to advance
-		if (sequence.length === 1 && sequence[0].type === 'Text' && nodes.length > 1) {
-			state.init.push(b.stmt(b.call('$.next')));
-		}
-
 		flush_sequence(sequence);
 	}
+
+	if (skipped > 1) {
+		skipped -= 1;
+		state.init.push(b.stmt(get_node(false)));
+	}
 }
 
 /**
- * @param {Expression} expression
- * @param {ComponentClientTransformState} state
- * @param {string} name
+ *
+ * @param {SvelteNode} node
  */
-function get_node_id(expression, state, name) {
-	let id = expression;
+function is_static_element(node) {
+	if (node.type !== 'RegularElement') return false;
+	if (node.fragment.metadata.dynamic) return false;
+
+	for (const attribute of node.attributes) {
+		if (attribute.type !== 'Attribute') {
+			return false;
+		}
+
+		if (is_event_attribute(attribute)) {
+			return false;
+		}
 
-	if (id.type !== 'Identifier') {
-		id = b.id(state.scope.generate(name));
+		if (attribute.value !== true && !is_text_attribute(attribute)) {
+			return false;
+		}
+
+		if (node.name === 'option' && attribute.name === 'value') {
+			return false;
+		}
 
-		state.init.push(b.var(id, expression));
+		if (node.name.includes('-')) {
+			return false; // TODO this feels unnecessary, but tests break
```

Comment:
It's not unnecessary, because we're setting all custom element attributes through their properties ([look at the code output](https://svelte-5-preview.vercel.app/#H4sIAAAAAAAAChXIQQqAIBQFwKvIWyvtpYTOUS00fyBYilok4t2j5UzD4TxlyKXh0idBYo4RHKXGH_khXwgcOdxp_2fUwrCX1Ulnq8ZBC6PWCxxnsO5wZCFLuqlv_QN4-tnZWwAAAA==))
```suggestion
			return false; // we're setting all attributes on custom elements through properties
```